src/languages/javascript.js

```diff
@@ -63,29 +63,36 @@ export default function(hljs) {
     literal: ECMAScript.LITERALS.join(" "),
     built_in: ECMAScript.BUILT_INS.join(" ")
   };
-  const nonDecimalLiterals = (prefixLetters, validChars) =>
-    `\\b0[${prefixLetters}][${validChars}]([${validChars}_]*[${validChars}])?n?`;
-  const noLeadingZeroDecimalDigits = /[1-9]([0-9_]*\d)?/;
-  const decimalDigits = /\d([0-9_]*\d)?/;
-  const exponentPart = regex.concat(/[eE][+-]?/, decimalDigits);
+
+  // https://tc39.es/ecma262/#sec-literals-numeric-literals
+  const decimalDigits = '[0-9](_?[0-9])*';
+  const frac = `\\.(${decimalDigits})`;
+  // DecimalIntegerLiteral, including Annex B NonOctalDecimalIntegerLiteral
+  // https://tc39.es/ecma262/#sec-additional-syntax-numeric-literals
+  const decimalInteger = `0|[1-9](_?[0-9])*|0[0-7]*[89][0-9]*`;
   const NUMBER = {
     className: 'number',
     variants: [
-      { begin: nonDecimalLiterals('bB', '01') }, // Binary literals
```

Comment:
I just wrote these beautiful abstractions not so long ago... 😐   Not saying change it, just shedding a quick tear.